### PR TITLE
[BugFix] Use correct logprobs for `logprob_token_ids`

### PR DESCRIPTION
**Pull request by njhill**
When gathering the logprobs for the `logprob_token_ids` option (used in generative scoring), we should honor the logprobs mode. This also avoids unnecessarily recomputing the logprobs from logits, and keeps the rank computation consistent with the top_k logprobs case.

Note this is a MRV1-specific fix. MRV2 does not have this issue.

### CODE REVIEW

**Review by yewentao256 — APPROVED** [review]
LGTM, thanks for the work!
